url/url.go

```
@@ -0,0 +1,339 @@
+	allowedSchemes := false
+	for _, scheme := range allowedReqSchemes {
+		if scheme == "" {
+			return false
+		}
+		if parsedURL.Scheme == scheme {
+			allowedSchemes = true
+			break
+		}
+	}
+
+	if !allowedSchemes {
```

golangci-lint

S1008: should use 'return allowedSchemes' instead of 'if !allowedSchemes { return false }; return true'

(gosimple)